dao/src/main/java/com/epam/esm/constants/ApplicationConstants.java

```diff
@@ -1,121 +1,117 @@
-    public final static String TAG_TABLE_NAME = "tag";
     public final static String CERTIFICATE_TAGS_TABLE_NAME = "certificates_tags";
-
-    //joins
-    public final static String GET_ALL_GC_BY_TAG_NAME = "SELECT gc.id, gc.name, gc.description, gc.price, gc.duration," +
-            " create_date, last_update_date FROM " + GIFT_CERTIFICATE_TABLE_NAME + " gc JOIN " + CERTIFICATE_TAGS_TABLE_NAME
-            + " ct on gc.id = ct.certificate_id JOIN " + TAG_TABLE_NAME + " t on t.id = ct.tag_id WHERE t.name = ?";
-    public final static String GET_ALL_TAG_BY_CERTIFICATE_ID = "SELECT t.id, t.name FROM " + TAG_TABLE_NAME +
-            " t JOIN certificates_tags ct on t.id = ct.tag_id WHERE ct.certificate_id = ?";
-
-    //gift_certificate table
-    public final static String GET_ALL_GC_QUERY = "SELECT id, name, description, price, duration," +
-            "create_date, last_update_date FROM " + GIFT_CERTIFICATE_TABLE_NAME;
-    public final static String GET_BY_ID_GC_QUERY = "SELECT id, name, description, price, duration, create_date," +
-            " last_update_date FROM " + GIFT_CERTIFICATE_TABLE_NAME + " WHERE id = ?";
-    public final static String SAVE_GC_QUERY = "INSERT INTO " + GIFT_CERTIFICATE_TABLE_NAME + " (name, description, price, duration) " + "VALUES (?,?,?,?)";
-    public final static String UPDATE_GC_QUERY = "UPDATE " + GIFT_CERTIFICATE_TABLE_NAME + " SET name = ?, description = ?, " +
-            "price = ?, duration = ? WHERE id = ?";
-    public final static String DELETE_GC_QUERY = "DELETE FROM " + GIFT_CERTIFICATE_TABLE_NAME + " WHERE id = ?";
-
-    public final static String GET_BY_NAME_GC_QUERY = "SELECT id, name, description, price, duration, create_date," +
-            " last_update_date FROM " + GIFT_CERTIFICATE_TABLE_NAME + " WHERE name = ?";
-
-    //tag table
-    public final static String GET_ALL_TAG_QUERY = "SELECT id, name FROM " + TAG_TABLE_NAME;
-    public final static String GET_BY_ID_TAG_QUERY = "SELECT id, name FROM " + TAG_TABLE_NAME + " WHERE id = ?";
-    public final static String SAVE_TAG_QUERY = "INSERT INTO " + TAG_TABLE_NAME + " (name) " +
-            "VALUES (?)";
-    public final static String DELETE_TAG_QUERY = "DELETE FROM " + TAG_TABLE_NAME + " WHERE id = ?";
-    public final static String GET_BY_NAME_TAG_QUERY = "SELECT id, name FROM " + TAG_TABLE_NAME + " WHERE name = ?";
-
-    //certificate_tags table
-    public final static String SAVE_CERTIFICATE_TAGS_QUERY = "INSERT INTO " + CERTIFICATE_TAGS_TABLE_NAME  + " VALUES (?,?)";
+    public final static String USERS_ORDERS_TABLE_NAME = "users_orders";
+
+    //GiftCertificate queries
+    public final static String FIND_GC_BY_NAME_QUERY = "SELECT gc FROM GiftCertificate gc WHERE gc.name = :name";
+
+    //Tag queries
+    public final static String GET_TAG_BY_NAME = "SELECT t FROM Tag t WHERE t.name =:name";
+    public final static String GET_MOST_WIDELY_USED_TAG = "SELECT t.id, t.name FROM tag t " +
```

Comment:
The high amount of JOINs. Optimize it

Reply:
I got rid of several redundant joins. Commit 7f10541.